tests/e2e/helper.py

```diff
@@ -21,3 +32,73 @@ def create_files(run_dir, *files):
 
         makedirs(parent_dir, exist_ok=True)
         open(full_path, encoding="utf-8", mode="a").close()
+
+
+def cleanup_local_test_files(*run_dirs) -> None:
+    """
+    Clean up the test files and logs etc.
+
+    To be called from TearDownClass to revert the test set up state.
+    """
+    for run_dir in run_dirs:
+        shutil.rmtree(Path(run_dir).parent)
+
+    for log_file in glob(path.join(TEST_DATA_DIR, "logs", "*log*")):
+        remove(log_file)
+
+    for log_file in glob(path.join(TEST_DATA_DIR, "logs/uploads/", "*log*")):
+        remove(log_file)
+
+    remove(path.join(TEST_DATA_DIR, "test_config.json"))
+
+
+def cleanup_remote_files(remote_path) -> None:
+    """
+    Clean up the uploaded test files from the remote path in the test
+    S3 bucket
+
+    Parameters
+    ----------
+    remote_path : str
+        path where files were uploaded to
+    """
+    bucket = boto3.resource("s3").Bucket(S3_BUCKET)
+    objects = bucket.objects.filter(Prefix=remote_path)
+    objects = [{"Key": obj.key} for obj in objects]
+
+    if objects:
+        bucket.delete_objects(Delete={"Objects": objects})
+
+
+def read_upload_log() -> dict:
+    """
+    Read in the run upload log file
+
+    Returns
+    -------
+    dict
+        contents of run upload log file
+    """
+    with open(
+        path.join(TEST_DATA_DIR, "logs/uploads/run_1.upload.log.json"),
```

Comment:
wouldn't the run name for this change and not always be `run_1` ?

Reply:
ah, I see this is only called for the testing of the partial runs, so you wouldn't need to generalise it for multiple log.json files